### PR TITLE
Model updates

### PR DESCRIPTION
**Pull request by gregjohnso**

### CODE REVIEW

File: fnet/fnet_model.py [hunk 1]
```diff
@@ -20,13 +21,14 @@ def __init__(
         self.gpu_ids = [gpu_ids] if isinstance(gpu_ids, int) else gpu_ids
         
         self.criterion = criterion_fn()
-        self._init_model()
+        self._init_model(model_kwargs)
 
-    def _init_model(self):
+    def _init_model(self, model_kwargs):
```
Comment (fcollman) [inline fnet/fnet_model.py]: if you made this **model_kwargs then it would keep backward compatibility

File: fnet/fnet_model.py [hunk 2]
```diff
@@ -20,13 +21,14 @@ def __init__(
         self.gpu_ids = [gpu_ids] if isinstance(gpu_ids, int) else gpu_ids
         
         self.criterion = criterion_fn()
-        self._init_model()
+        self._init_model(model_kwargs)
```
Comment (fcollman) [inline fnet/fnet_model.py]: this could then be **model_kwargs

File: fnet/fnet_model.py [hunk 3]
```diff
@@ -67,10 +69,10 @@ def save_state(self, path_save):
         torch.save(self.get_state(), path_save)
         self.to_gpu(curr_gpu_ids)
 
-    def load_state(self, path_load, gpu_ids=-1):
+    def load_state(self, path_load, gpu_ids=-1, model_kwargs={}):
         state_dict = torch.load(path_load)
         self.nn_module = state_dict['nn_module']
-        self._init_model()
+        self._init_model(model_kwargs)
```
Comment (fcollman) [inline fnet/fnet_model.py]: would all the model_kwargs be better put with the state... why should i have to remember how i initialized the model when i'm reloading it?

File: fnet/nn_modules/fnet_nn_3d.py [hunk 4]
```diff
@@ -1,16 +1,15 @@
...
 # parent: v5_nn.py
 # changed from v5: removed ReLU at output
 
 class Net(torch.nn.Module):
-    def __init__(self):
+    def __init__(self, out_channels = 1):
         super().__init__()
         mult_chan = 32
         depth = 4
         self.net_recurse = _Net_recurse(n_in_channels=1, mult_chan=mult_chan, depth=depth)
```
Comment (fcollman) [inline fnet/nn_modules/fnet_nn_3d.py]: fwiw i would make all these parameters keyword arguments, that's what i've done.

File: fnet/utils/figures.py [hunk 5]
```diff
@@ -213,6 +213,9 @@ def time_series_to_img(im_path_list, window_position = None, window_size = None,
         for im_channel in im_t:
             im = imread(im_channel)
             
+            if len(im.shape) == 4:
+                im = im[:,32,:,:]
+            
             if im.shape[1] == 1:
```
Comment (fcollman) [inline fnet/utils/figures.py]: i'm not sure this line makes sense anymore

File: fnet/nn_modules/fnet_nn_3d.py (hunk 4 above)
Comment (counkomol) [reply fnet/nn_modules/fnet_nn_3d.py]: Let's keep the original `fnet_nn_3d.py ` without parameters. We can define a copy of it with all parameters specifiable.

File: fnet/fnet_model.py (hunk 2 above)
Comment (counkomol) [reply fnet/fnet_model.py]: I'm okay with the _init_model() always requiring a parameter. I think `model_kwargs` should be changed to `nn_kwargs` though. It's more accurate.

File: fnet/fnet_model.py (hunk 3 above)
Comment (counkomol) [reply fnet/fnet_model.py]: Agree, model/neural net hyperparameters should definitely go into the save state.

File: fnet/utils/figures.py (hunk 5 above)
Comment (counkomol) [reply fnet/utils/figures.py]: @gregjohnso Could you make the changes to `figures.py` and `utils.py`  a separate PR?

**Pull request closed without merging**